### PR TITLE
Test sharing with ocs api v2

### PR DESCRIPTION
**Pull request by individual-it**
## Description
sharing can be used via `v1.php` and via `v2.php`, we never tested it with OCS API v2, so this adds some tests and some more checks where applicable

~depends on #32052 & #32072~

## Related Issue
#32040

## Motivation and Context
having better test coverage

## How Has This Been Tested?
run tests locally, drone will do it again :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiSharing-v1/gettingShares.feature [hunk 1]
```diff
@@ -91,19 +106,25 @@ Feature: sharing
...
+		Examples:
+			|ocs_api_version|http_status_code|
+			|1              |200             |
+			|2              |404             |
 
-Scenario: Share of folder to a group, remove user from that group
-		Given user "user0" has been created
-		And user "user1" has been created
+	Scenario: Share of folder to a group, remove user from that group
+		Given using API version "<ocs_api_version>"
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/gettingShares.feature]: Looks like this should be a scenario outline?

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 2]
```diff
@@ -115,12 +127,11 @@ Feature: sharing
 			| shareType   | 0      |
 			| permissions | 1      |
 		When user "user1" deletes file "/shared/shared_file.txt" using the API
-		Then the HTTP status code should be "403"
+		And the HTTP status code should be "403"
 		And as "user1" the file "/shared/shared_file.txt" should exist
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: This change not needed

File: tests/acceptance/features/apiSharing-v1/uploadToShare.feature [hunk 3]
```diff
@@ -116,7 +107,6 @@ Feature: sharing
 		Then the content of file "/FOLDER/test.txt" for user "user0" should be "test"
 
 	Scenario: Uploading file to a user read/write share folder works
-		Given user "user0" has been created
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/uploadToShare.feature]: Needs a "given" belowqq

**Review by phil-davis — APPROVED** [review]
Just a few little things you can fix.

**Comment by phil-davis** [comment]
In some scenarios you still only doOCS API version 1.
Can you make some comment about why those scenarios are not a scenario outline that also has version 2.

File: tests/acceptance/features/apiSharing-v1/gettingShares.feature (hunk 1 above)
Comment (individual-it) [reply tests/acceptance/features/apiSharing-v1/gettingShares.feature]: no that has actually only to test API v1

**Comment by individual-it** [comment]
- Wherever the return codes (OCS or HTTP) of the sharing API are tested, both versions are checked 
- Whenever a step calls the sharing API directly (e.g. ` 		When user "user0" sends HTTP method "GET" to API endpoint "/apps/files_sharing/api/v1/shares?path=PARENT&subfiles=true"`, both versions are checked
- When steps do not specify how exactly the share was creates only one version is checked, because the step says "using the API" and should not care about which API is used
- Whenever the return codes of some webdav action are checked also only one version of the ocs api is checked assuming it does not matter how the sharing is done

**Pull request merged**